src/server/game/Entities/Creature/Creature.h

```diff
@@ -45,6 +45,8 @@ class Creature : public Unit, public GridObject<Creature>, public MovableMapObje
 
     void DisappearAndDie();
 
+    bool isVendorWithIconSpeak() const;
```

Comment:
please consider using `[[nodiscard]]` where applicable. This looks a good candidate for it